nbsphinx_link/__init__.py

```diff
@@ -19,53 +19,91 @@
 Further keys might be added in the future.
 """
 
-import errno
 import json
 import os
 import shutil
 
 from docutils import io, nodes, utils
 from docutils.utils.error_reporting import SafeString, ErrorString
-from docutils.nodes import document
+import docutils  # noqa: F401
 from nbsphinx import NotebookParser, NotebookError, _ipynbversion
 import nbformat
 from sphinx.util.logging import getLogger
 
 from ._version import __version__
 
 
-def copy_directory(src, dest):
+def register_dependency(file_path, document):
     """
-    Copies a directory or file from the path ``src`` to ``dest``.
-    Originally from: https://www.pythoncentral.io/how-to-recursively-copy-a-directory-folder-in-python/
-    Which is licensed under the MIT License: https://opensource.org/licenses/MIT
+    Registers files as dependency, so sphinx rebuilds the docs
+    when they changed.
 
+    Parameters
+    ----------
+    file_path : str
+        [description]
+    document: docutils.nodes.document
+        Parsed document instance.
+    """
+    document.settings.record_dependencies.add(file_path)
+    document.settings.env.note_dependency(file_path)
+
+
+def copy_file(src, dest):
+    """
+    Copies a singe file from ``src`` to ``dest``.
+
+    Parameters
+    ----------
+    src : str
+        Path to the source file.
+    dest : str
+        Path to the destination file.
+    """
+    logger = getLogger(__name__)
```

Comment:
We should use the Sphinx logger instead.

Reply:
I.e. `sphinx.util.logging.getLogger(__name__)`, not the deprecated logging API.

Reply:
Its is ``from sphinx.util.logging import getLogger`` or do you have a different logger in mind?

Reply:
I already commented here, but I think you might have missed the second comment? :)

Reply:
Looks like we talk about the same ``logger``, so it should be fine. :)